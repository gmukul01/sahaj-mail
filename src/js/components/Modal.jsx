import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

const Modal = props => {
	const { showModal, onCloseModal, header } = props;
	const modalClassName = classNames('modal', { open: showModal });

	return (
		<div className={modalClassName}>
			<div className="modal-content">
				<div className="modal-header">
					<h3 className="title">{header}</h3>
					<button className="modal-close" onClick={onCloseModal}>
						Ã—
					</button>
				</div>
				<div className="modal-body">{props.children}</div>
			</div>
		</div>
	);
};

Modal.propTypes = {
	showModal: PropTypes.bool,
	onCloseModal: PropTypes.func,
	header: PropTypes.string
};

export default React.memo(Modal);
