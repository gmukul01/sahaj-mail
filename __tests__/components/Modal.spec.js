import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Modal from 'components/Modal';

describe('Modal Component', () => {
	const onCloseHandler = jest.fn();

	it('should match snapshot when showModal is true', () => {
		const modal = renderer.create(<Modal showModal header="Dummy Header" onCloseModal={onCloseHandler} />).toJSON();
		expect(modal).toMatchSnapshot();
	});

	it('should match snapshot when showModal is false', () => {
		const modal = renderer.create(<Modal showModal={false} header="Dummy Header" onCloseModal={onCloseHandler} />).toJSON();
		expect(modal).toMatchSnapshot();
	});

	it('should call onClick Handler on onClick event', () => {
		const modal = shallow(<Modal id={1} selected onCloseModal={onCloseHandler} />);
		modal.find('button').simulate('click');
		expect(onCloseHandler.mock.calls.length).toBe(1);
	});
});
