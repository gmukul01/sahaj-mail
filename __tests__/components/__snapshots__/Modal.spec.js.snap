// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component should match snapshot 1`] = `
Array [
  <input
    className="styled-checkbox"
    defaultChecked={true}
    id="checkbox-1"
    onChange={[MockFunction]}
    type="checkbox"
    value={1}
  />,
  <label
    htmlFor="checkbox-1"
  />,
]
`;

exports[`Modal Component should match snapshot 1`] = `
<div
  className="modal open"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <h3
        className="title"
      >
        Dummy Header
      </h3>
      <button
        className="modal-close"
      >
        ×
      </button>
    </div>
    <div
      className="modal-body"
    />
  </div>
</div>
`;

exports[`Modal Component should match snapshot when showModal is false 1`] = `
<div
  className="modal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <h3
        className="title"
      >
        Dummy Header
      </h3>
      <button
        className="modal-close"
        onClick={[MockFunction]}
      >
        ×
      </button>
    </div>
    <div
      className="modal-body"
    />
  </div>
</div>
`;

exports[`Modal Component should match snapshot when showModal is true 1`] = `
<div
  className="modal open"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <h3
        className="title"
      >
        Dummy Header
      </h3>
      <button
        className="modal-close"
        onClick={[MockFunction]}
      >
        ×
      </button>
    </div>
    <div
      className="modal-body"
    />
  </div>
</div>
`;

exports[`Modal Component should match snapshot when showModal is true 2`] = `
<div
  className="modal open"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <h3
        className="title"
      >
        Dummy Header
      </h3>
      <button
        className="modal-close"
        onClick={[MockFunction]}
      >
        ×
      </button>
    </div>
    <div
      className="modal-body"
    />
  </div>
</div>
`;
